### DISCUSSION TITLE
Is there a way to create a pie chart from columns of one row ?

### DISCUSSION THREAD

**Discussion by estelle-d** [Budibase/budibase#9001] [Help]
Hello,

I would like to create a pie chart for the data within one row:
**score | serie 1 | serie 2 | serie 3 | serie 4**

the score is the full circle and the pie are the "serie **n**"

that chart will go in the **/screen/:id screen** 

Is it possible?
Thank you
Estelle

**Comment by estelle-d** [comment]
To give you the context, one row represente a match with the score and number one point in each series, and I would like to create a pie chart for each match without having to create a new table foe each match.

Thank's
Estelle

**Comment by melohagan** [comment]
Hey @estelle-d 

If you are using an external datasource, then you can use a [custom query](https://docs.budibase.com/docs/custom-queries) with a [transformer](https://docs.budibase.com/docs/chart#generating-the-chart-schema) to get the data in the shape you need.

If you are using the internal Budibase DB, then there is no way to use multiple column values as segments in the Pie chart. Instead you could use the JSON data type like so:

![Screenshot 2022-12-20 at 10 18 22](https://user-images.githubusercontent.com/101575380/208643196-0ff6ef92-df6d-4027-a565-a67a898f8ed8.png)

![Screenshot 2022-12-20 at 10 18 41](https://user-images.githubusercontent.com/101575380/208643257-1f5e6cb7-1c9d-46a8-851f-124e227af25e.png)

![Screenshot 2022-12-20 at 10 19 07](https://user-images.githubusercontent.com/101575380/208643336-4499fa11-3f44-40d8-ba92-4011646c97f4.png)

![Screenshot 2022-12-20 at 10 19 47](https://user-images.githubusercontent.com/101575380/208643487-5bfa992b-817d-40e3-b270-2dbb49c273be.png)

![Screenshot 2022-12-20 at 10 20 56](https://user-images.githubusercontent.com/101575380/208643726-ddb54d03-3415-4d99-a59f-0f27d766a515.png)

**Comment by estelle-d** [comment]
Thank you very much!!!! I thought it may have been possible with an external DB but I couldn't found the info, it's a great news!!!!
I am going to try it out soon... I have reach my passe limitation for this month :(

**Reply by phamalda** [reply]
Hi, @estelle-d. I know it's been a long time since you posted this thread. Just wanted to check in to see if you were able to get this to work in creating that pie chart. Thanks.
